src/error.rs

```diff
@@ -233,6 +239,11 @@ pub(crate) fn url_bad_scheme(url: Url) -> Error {
     Error::new(Kind::Builder, Some("URL scheme is not allowed")).with_url(url)
 }
 
+#[cfg(feature = "__tls")]
+pub(crate) fn unknown_preconfigured_tls() -> Error {
```

Comment:
Instead of creating a new `Kind`, I'd just consider this a `Kind::Builder` error.

Reply:
Makes sense yeah. Updated